rclpy/CMakeLists.txt

```diff
@@ -83,6 +83,13 @@ endmacro()
 
 call_for_each_rmw_implementation(target)
 
+set(RMW_IMPLEMENTATIONS "")
+
+macro(rmw_list)
+  set(RMW_IMPLEMENTATIONS "${RMW_IMPLEMENTATIONS} ${rmw_implementation}")
+endmacro()
+call_for_each_rmw_implementation(rmw_list)
```

Comment:
There is much simpler API for this: https://github.com/ros2/rmw/blob/master/rmw_implementation_cmake/cmake/get_available_rmw_implementations.cmake